src/value_generators/test/DocGenYamlTestCaseRunner.cpp

```diff
@@ -32,21 +33,33 @@ class ValGenTestCase {
             msg << "Can't have ThenReturns and ThenThrows in '" << toString(node) << "'";
             throw std::invalid_argument(msg.str());
         }
+        if (_thenExecuteAndIgnore && _expectedExceptionMessage) {
```

Comment:
I think this is still leaving open the possibility of someone putting `ThenReturns` and also `ThenExecuteAndIgnore`, which doesn't seem to make sense. Should we combine this with the previous if-statement and check that only one of the three is set?

Reply:
Yeah I thought about this too and couldn't figure out a nifty way of doing a 3-way XOR (Though https://stackoverflow.com/questions/6228252/xor-with-3-values has some fun ideas of assigning 1 to true values and 0 to false values and throwing if the sum != 1) 

I went ahead and added a 3rd if statement for `ThenReturns` and `ThenExecuteAndIgnore` which also does the job but is a bit clunky